graylog2-server/src/main/java/org/graylog/events/processor/EventProcessorExecutionMetrics.java

```diff
@@ -0,0 +1,103 @@
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.graylog.events.processor;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import org.graylog2.shared.metrics.MetricUtils;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Locale;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+@Singleton
+public class EventProcessorExecutionMetrics {
+    private final MetricRegistry metricRegistry;
+
+    @Inject
+    public EventProcessorExecutionMetrics(MetricRegistry metricRegistry) {
+        this.metricRegistry = metricRegistry;
+    }
+
+    private enum Fields {
+        EXECUTION_COUNT(Counter::new),
+        EXECUTION_SUCCESSFUL(Counter::new),
+        EXECUTION_EXCEPTION(Counter::new),
+        EXECUTION_TIME(Timer::new),
+        EVENTS_CREATED(Meter::new);
+
+        private final Supplier<Metric> type;
+        Fields(Supplier<Metric> type) {
+            this.type = type;
+        }
+    }
+
+    void registerEventProcessor(EventProcessor eventProcessor, String definitionId) {
+        for (Fields field: Fields.values() ) {
+            final String name = getNameforField(eventProcessor, definitionId, field);
+            MetricUtils.safelyRegister(metricRegistry, name, field.type.get());
+        }
+    }
+    void recordExecutionTime(EventProcessor eventProcessor, String definitionId, long time, TimeUnit unit) throws Exception {
+        final String name = getNameforField(eventProcessor, definitionId, Fields.EXECUTION_TIME);
+        MetricUtils.getOrRegister(metricRegistry, name, new Timer()).update(time, unit);
+    }
+
+    void recordExecutions(EventProcessor eventProcessor, String definitionId) {
+        final String name = getNameforField(eventProcessor, definitionId, Fields.EXECUTION_COUNT);
+        MetricUtils.getOrRegister(metricRegistry, name, new Counter()).inc();
+    }
+    void recordSuccess(EventProcessor eventProcessor, String definitionId) {
+        final String name = getNameforField(eventProcessor, definitionId, Fields.EXECUTION_SUCCESSFUL);
+        MetricUtils.getOrRegister(metricRegistry, name, new Counter()).inc();
+    }
+    void recordException(EventProcessor eventProcessor, String definitionId) {
+        final String name = getNameforField(eventProcessor, definitionId, Fields.EXECUTION_EXCEPTION);
+        MetricUtils.getOrRegister(metricRegistry, name, new Counter()).inc();
+    }
+    void recordCreatedEvents(EventProcessor eventProcessor, String definitionId, int count) {
+        final String name = getNameforField(eventProcessor, definitionId, Fields.EVENTS_CREATED);
+        MetricUtils.getOrRegister(metricRegistry, name, new Meter()).mark(count);
+    }
+
+    private static String getNameforField(EventProcessor eventProcessor, String definitionId, Fields field) {
```

Comment:
How about renaming this for camel-case consistency?

```suggestion
    private static String getNameForField(EventProcessor eventProcessor, String definitionId, Fields field) {
```